build_tools/cmake/static_linker_test.c.in

```diff
@@ -0,0 +1,217 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// clang-format off
```

Comment:
This is a template for a C++ file, so having IDEs and such treat it as C++ helps with syntax highlighting and formatting for the non-templated parts is still nice. In fact, I'm considering renaming these to `.in.cc` / `.template.cc` / etc. so that we get those language features automatically.